compiler-plugin/src/main/kotlin/it/unibo/collektive/frontend/MissingAlignExtension.kt

```diff
@@ -0,0 +1,108 @@
+    containingElements.any { (it as? FirSimpleFunction)?.receiverParameter?.isAggregate(session) == true }
+
+/**
+ * Checker that looks for aggregate functions called inside a loop without an explicit align operation.
+ */
+object NoAlignInsideALoop : FirFunctionCallChecker(MppCheckerKind.Common) {
+
+    private fun CheckerContext.isInsideALoopWithoutAlignedOn(): Boolean {
+        val loopElementIndex = containingElements.indexOfLast { it is FirWhileLoop }
+        val functionDeclarationIndex = containingElements.dropLast(1).indexOfLast {
+            (it as? FirSimpleFunction)?.receiverParameter?.isAggregate(session) == true
+        }
+        if (loopElementIndex == -1 || functionDeclarationIndex == -1) {
+            return false
+        }
+        return loopElementIndex > functionDeclarationIndex &&
+            containingElements
+                .drop(loopElementIndex)
+                .dropLast(1)
+                .filterIsInstance<FirFunctionCall>()
+                .map { it.calleeReference.name.asString() }
+                .none { it == "alignedOn" }
+    }
+
+    override fun check(
+        expression: FirFunctionCall,
+        context: CheckerContext,
+        reporter: DiagnosticReporter,
+    ) {
+        val calleeName = expression.calleeReference.name.identifier
+        if (expression.isAggregate(context.session) &&
+            context.isInsideALoopWithoutAlignedOn()
+        ) {
+            reporter.reportOn(
+                expression.calleeReference.source,
+                PluginErrors.DOT_CALL_WARNING,
+                "Warning: aggregate function \"$calleeName\" called inside a loop with no manual alignment operation",
+                context,
+            )
+        }
+    }
+}
+
+/**
+ * Checker that looks for usages of "align" and "dealign" methods inside Aggregate functions, generating a warning.
+ */
+object NoAlignOrDealign : FirFunctionCallChecker(MppCheckerKind.Common) {
+    override fun check(expression: FirFunctionCall, context: CheckerContext, reporter: DiagnosticReporter) {
+        val calleeName = expression.calleeReference.name.identifier
+        if ((calleeName == "align" || calleeName == "dealign") && context.isInsideAggregateFunction()) {
+            reporter.reportOn(
+                expression.calleeReference.source,
+                PluginErrors.DOT_CALL_WARNING,
+                "Warning: \"%s\" method should not be explicitly used".format(calleeName),
+                context,
+            )
+        }
+    }
+}
+
+/**
+ * Extension that adds a series of checkers that looks for missing align operations within the Collektive DSL.
+ */
+class MissingAlignExtension(session: FirSession) : FirAdditionalCheckersExtension(session) {
+    override val expressionCheckers: ExpressionCheckers = object : ExpressionCheckers() {
+        override val functionCallCheckers: Set<FirFunctionCallChecker>
+            get() = setOf(NoAlignInsideALoop, NoAlignOrDealign)
+    }
+}
```

Comment:
Unless all top level members can be private but one, I would split this into multiple files, possibly creating a new subpackage.